File: backend/src/models/UserLikeSongs.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IUserLikeSongs extends Document {
    userId: number;
    songId: string;
    isLiked: boolean;
}

const UserLikeSongsSchema: Schema = new Schema({
    userId: {
        type: Number,
        required: [true, '用户ID不能为空'],
        trim: true
    },
    songId: {
        type: String,
        required: [true, '歌曲ID不能为空'],
        trim: true
    },
    isLiked: {
        type: Boolean,
        default: false
    }
});

UserLikeSongsSchema.index({ userId: 1, songId: 1 }, { unique: true });

const UserLikeSongs = mongoose.model<IUserLikeSongs>('user_like_songs', UserLikeSongsSchema);
export default UserLikeSongs;

File: backend/java-backend/src/main/java/com/example/demo/model/mongo/UserLikeSongs.java
package com.example.demo.model.mongo;

import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.CompoundIndex;

@Data
@Document(collection = "user_like_songs")
@CompoundIndex(name = "userId_songId_unique", def = "{'userId': 1, 'songId': 1}", unique = true)
public class UserLikeSongs {

    @Id
    private String id;

    private Integer userId;
    // songId 字段类型为 ObjectId，与查询类型保持一致，避免 MongoDB 类型不匹配
    private ObjectId songId;
    private Boolean isLiked;
}
